torch/_dynamo/variables/builder.py

```diff
@@ -440,6 +440,31 @@ def index_source(key):
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
         # NB: These can't be put in type_dispatch, they have to run later
+        elif value in {
```

Comment:
also assumption of hashable in this tree is unsafe, can't just `value in` - Please add a check here that this is a function, or do the comparison directly, or make a list.

Reply:
yep good point. i will do this part another way